sdv/metadata/single_table.py

```diff
@@ -338,6 +338,9 @@ def _detect_columns(self, data):
                 datetime_format = get_datetime_format(column_data.iloc[:100])
                 column_dict['datetime_format'] = datetime_format
 
+            if self.primary_key is None and sdtype == 'id':
```

Comment:
I set it here to avoid having to loop through the columns again. Another option could be to track a temporary variable instead and then check/set the primary key outside of the loop, but I'm not sure how much clearer that would be.